### PR TITLE
add plan finder to integration, send params for backwards compatibility

### PR DESCRIPTION
**Pull request by jdelbick**
## Description
Made some of these changes while testing out plan finder e2e. we don't have to merge this if there are any concerns
- add plan finder to pipeline integration in the SDK instead of adding to plan directly
- add `png-png` test case that does not rely on imagemagickTransformer (no postprocessing)
- switch from symlinking to copying because there were some issues in the tests
- pass along `input.params` to workerCB for backwards compatibility: third parameter in workerCB is params: https://github.com/adobe/asset-compute-sdk#params

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -75,12 +75,10 @@ function worker(renditionCallback, options = {}) {
                 path: `${testDir}/${params.source}`
             };
         }
-
         // note, we will need to add support for multiple renditions
         // note, we have to integrate options into our plan and support everything we support now
-        plan.add(transformer.name, { input, output: params.renditions[0] });
-
-        console.log('Plan:', plan.toObject());
+        // plan.add(transformer.name, { input, output: params.renditions[0] });
```
Comment (tmathern) [inline lib/api.js:80]:
```suggestion
        
```

File: lib/worker-transformer.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ class WorkerCallbackTransformer extends Transformer {
     }
 
     compute(input, output) {
-        return this._callback(input, output);
+        return this._callback(input, output, input.params);
```
Comment (tmathern) [inline lib/worker-transformer.js:29]: Not in this PR - but when constructing the class, there should be a check that `this._callback` is a function.
```suggestion
        // TODO: Verify `this._callback` is a function.
        return this._callback(input, output, input.params);
```

File: test-worker/test/asset-compute/worker/png-png/validate [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# Copyright 2020 Adobe. All rights reserved.
```
Comment (tmathern) [inline test-worker/test/asset-compute/worker/png-png/validate:3]:
```suggestion
# Copyright 2021 Adobe. All rights reserved.
```

File: test-worker/worker.js [hunk 4]
```diff
@@ -60,7 +60,8 @@ exports.main = worker(async (source, rendition) => {
         console.log(`[test worker] copying source to rendition: ${source.path} to ${rendition.path}`);
         // simple case where post processing just runs on the source files for basic tests
         // symlink source to rendition to transfer 1:1
-        await fs.symlink(source.path, rendition.path);
+        // await fs.symlink(source.path, rendition.path);
```
Comment (tmathern) [inline test-worker/worker.js:63]:
```suggestion
        
```

**Review by tmathern — APPROVED** [review]
Is there a reason not to use a symlink for the `png` file in `png-png` test?
A similar looking file is available already at a higher folder level (https://github.com/adobe/asset-compute-sdk/blob/778d09a2f034dfd3af699b5624f37c07aa57c63b/test-worker/test/asset-compute/worker/file.png).

I have the same question/comment about the validation script in the same test.

**Comment by jdelbick** [comment]
@tmathern not symlinking was a copy/paste mistake. fixed it

**Pull request merged**